Can't build with `linuxPackages_grsec_desktop_4_5`

## Issue description
End of the build output:
```
shrinking RPATHs of ELF executables and libraries in 
find: cannot search `': No such file or directory
building path(s) ‘/nix/store/k0a3gkkns48l0xbp3d5a8hi84znzd8v0-unit-systemd-sysctl.service’
building path(s) ‘/nix/store/jvbqdrbkg7nk6avgsn822xc5b2ysh10y-broadcom-sta-6.30.223.248-4.5.6’
unpacking sources
Makefile
lib/
lib/LICENSE.txt
lib/wlc_hybrid.o_shipped
src/
src/common/
src/common/include/
src/common/include/proto/
src/common/include/proto/802.11.h
src/common/include/proto/bcmevent.h
src/common/include/proto/bcmeth.h
src/common/include/proto/ethernet.h
src/common/include/proto/ieee80211_radiotap.h
src/common/include/proto/802.1d.h
src/common/include/proto/wpa.h
src/common/include/proto/bcmip.h
src/include/
src/include/packed_section_end.h
src/include/bcmutils.h
src/include/packed_section_start.h
src/include/bcmdefs.h
src/include/epivers.h
src/include/pcicfg.h
src/include/bcmcrypto/
src/include/bcmcrypto/tkhash.h
src/include/siutils.h
src/include/wlioctl.h
src/include/linux_osl.h
src/include/bcmendian.h
src/include/linuxver.h
src/include/osl.h
src/include/typedefs.h
src/shared/
src/shared/bcmwifi/
src/shared/bcmwifi/include/
src/shared/bcmwifi/include/bcmwifi_rates.h
src/shared/bcmwifi/include/bcmwifi_channels.h
src/shared/linux_osl.c
src/wl/
src/wl/sys/
src/wl/sys/wl_linux.c
src/wl/sys/wlc_key.h
src/wl/sys/wlc_utils.h
src/wl/sys/wl_iw.c
src/wl/sys/wlc_types.h
src/wl/sys/wl_export.h
src/wl/sys/wlc_ethereal.h
src/wl/sys/wl_iw.h
src/wl/sys/wlc_wowl.h
src/wl/sys/wl_cfg80211_hybrid.h
src/wl/sys/wl_linux.h
src/wl/sys/wl_cfg80211_hybrid.c
src/wl/sys/wl_dbg.h
src/wl/sys/wlc_pub.h
patching sources
applying patch /nix/store/1x0famv0lh5sm6snqp24yx2zl9wf6nsz-license.patch
patching file src/wl/sys/wl_linux.c
Hunk #1 succeeded at 148 (offset -23 lines).
applying patch /nix/store/yykwplnyfp9l3agnqahchrx17qr2w2r8-linux-recent.patch
patching file src/wl/sys/wl_cfg80211_hybrid.c
patching file src/wl/sys/wl_dbg.h
patching file src/wl/sys/wl_linux.c
Hunk #1 succeeded at 880 (offset 2 lines).
Hunk #2 succeeded at 1272 (offset 2 lines).
Hunk #3 succeeded at 1310 (offset 2 lines).
Hunk #4 succeeded at 1659 (offset 2 lines).
Hunk #5 succeeded at 2161 (offset 2 lines).
Hunk #6 succeeded at 2459 (offset 2 lines).
Hunk #7 succeeded at 2541 (offset 2 lines).
Hunk #8 succeeded at 2672 (offset 2 lines).
Hunk #9 succeeded at 3000 (offset 2 lines).
Hunk #10 succeeded at 3234 (offset 2 lines).
Hunk #11 succeeded at 3313 (offset 2 lines).
Hunk #12 succeeded at 3327 (offset 2 lines).
Hunk #13 succeeded at 3356 (offset 2 lines).
applying patch /nix/store/2m9lvi88kwm3n1mka728zpva8nb54ybs-gcc.patch
patching file Makefile
configuring
grep: : No such file or directory
grep: : No such file or directory
no configure script, doing nothing
building
make flags: SHELL=/nix/store/i7hx6w6zy3bv53f2xm1r23ya8qbzn4is-bash-4.3-p42/bin/bash KBASE=/nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec   
KBUILD_NOPEDANTIC=1 make -C /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/build M=`pwd`
make[1]: Entering directory '/nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/build'
++ dirname /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/scripts/gcc-plugin.sh
+ srctree=/nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/scripts
++ gcc -print-file-name=plugin
+ gccplugins_dir=/nix/store/c7ipds48nb7sfzhb7vqp26rrllirxwxv-gcc-5.3.0/lib/gcc/x86_64-unknown-linux-gnu/5.3.0/plugin
++ g++ -E -x c++ - -o /dev/null -I/nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/scripts/gcc-plugins -I/nix/store/c7ipds48nb7sfzhb7vqp26rrllirxwxv-gcc-5.3.0/lib/gcc/x86_64-unknown-linux-gnu/5.3.0/plugin/include
+ plugincc='In file included from /nix/store/c7ipds48nb7sfzhb7vqp26rrllirxwxv-gcc-5.3.0/lib/gcc/x86_64-unknown-linux-gnu/5.3.0/plugin/include/gcc-plugin.h:28:0,
                 from /nix/store/c7ipds48nb7sfzhb7vqp26rrllirxwxv-gcc-5.3.0/lib/gcc/x86_64-unknown-linux-gnu/5.3.0/plugin/include/plugin.h:23,
                 from /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/scripts/gcc-plugins/gcc-common.h:8,
                 from <stdin>:1:
/nix/store/c7ipds48nb7sfzhb7vqp26rrllirxwxv-gcc-5.3.0/lib/gcc/x86_64-unknown-linux-gnu/5.3.0/plugin/include/system.h:662:17: fatal error: gmp.h: No such file or directory
compilation terminated.'
+ '[' 1 -ne 0 ']'
+ exit 1
scripts/Makefile.gcc-plugins:61: warning, your gcc installation does not support plugins, perhaps the necessary headers are missing?
scripts/Makefile.gcc-plugins:65: PAX_MEMORY_STACKLEAK and other features will be less secure
CFG80211 API is prefered for this kernel version
Using CFG80211 API
  LD      /tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/built-in.o
  CC [M]  /tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/shared/linux_osl.o
Function osl_malloc is missing from the size_overflow hash table +osl_malloc+fndecl+2+46071+
  CC [M]  /tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_linux.o
Function len is missing from the size_overflow hash table +len+wl_ioctl+0+40939+
  CC [M]  /tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_iw.o
  CC [M]  /tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.o
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c: In function 'wl_init_eloop_handler':
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2512:9: warning: passing argument 1 of 'memset' discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  memset(el, 0, sizeof(*el));
         ^
In file included from /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/arch/x86/include/asm/string.h:4:0,
                 from /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/include/linux/string.h:18,
                 from /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/include/linux/dynamic_debug.h:111,
                 from /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/include/linux/printk.h:277,
                 from /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/include/linux/kernel.h:13,
                 from /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/include/linux/list.h:8,
                 from /nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/include/linux/module.h:9,
                 from /tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/include/linuxver.h:40,
                 from /tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:26:
/nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/arch/x86/include/asm/string_64.h:55:7: note: expected 'void *' but argument is of type 'const struct wl_cfg80211_event_loop *'
 void *memset(void *s, int c, size_t n);
       ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2513:35: error: assignment of read-only location 'el->handler[26]'
  el->handler[WLC_E_SCAN_COMPLETE] = wl_notify_scan_status;
                                   ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2514:26: error: assignment of read-only location 'el->handler[1]'
  el->handler[WLC_E_JOIN] = wl_notify_connect_status;
                          ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2515:27: error: assignment of read-only location 'el->handler[2]'
  el->handler[WLC_E_START] = wl_notify_connect_status;
                           ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2516:26: error: assignment of read-only location 'el->handler[16]'
  el->handler[WLC_E_LINK] = wl_notify_connect_status;
                          ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2517:31: error: assignment of read-only location 'el->handler[18]'
  el->handler[WLC_E_NDIS_LINK] = wl_notify_connect_status;
                               ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2518:30: error: assignment of read-only location 'el->handler[0]'
  el->handler[WLC_E_SET_SSID] = wl_notify_connect_status;
                              ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2519:34: error: assignment of read-only location 'el->handler[12]'
  el->handler[WLC_E_DISASSOC_IND] = wl_notify_connect_status;
                                  ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2520:32: error: assignment of read-only location 'el->handler[6]'
  el->handler[WLC_E_DEAUTH_IND] = wl_notify_connect_status;
                                ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2521:26: error: assignment of read-only location 'el->handler[19]'
  el->handler[WLC_E_ROAM] = wl_notify_roaming_status;
                          ^
/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.c:2522:31: error: assignment of read-only location 'el->handler[17]'
  el->handler[WLC_E_MIC_ERROR] = wl_notify_mic_status;
                               ^
/nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/scripts/Makefile.build:258: recipe for target '/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.o' failed
make[4]: *** [/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta/src/wl/sys/wl_cfg80211_hybrid.o] Error 1
/nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/source/Makefile:1409: recipe for target '_module_/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta' failed
make[3]: *** [_module_/tmp/nix-build-broadcom-sta-6.30.223.248-4.5.6.drv-0/broadcom-sta] Error 2
Makefile:146: recipe for target 'sub-make' failed
make[2]: *** [sub-make] Error 2
Makefile:24: recipe for target '__sub-make' failed
make[1]: *** [__sub-make] Error 2
make[1]: Leaving directory '/nix/store/pa0m9qa759lv3l3gdn4cyq997pxqwdvw-linux-4.5.6-dev/lib/modules/4.5.6-grsec/build'
Makefile:142: recipe for target 'all' failed
make: *** [all] Error 2
builder for ‘/nix/store/49g3wcbb6pyawdixh160q3vslhd3fmdm-broadcom-sta-6.30.223.248-4.5.6.drv’ failed with exit code 2
building path(s) ‘/nix/store/hax658x97qjrksf87a48zrvvn5i9spi1-firmware’
cannot build derivation ‘/nix/store/nw6j7yzmaxdvpaayj00c190zhlqbhmw4-kernel-modules.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/nhlbgqvndfj1xv8slz04z1bnzx1c788d-nixos-system-nixos-16.03.1083.d745044.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/nhlbgqvndfj1xv8slz04z1bnzx1c788d-nixos-system-nixos-16.03.1083.d745044.drv’ failed

```

### Steps to reproduce
Set in the /etc/nixos/configuration.nix
```
boot.kernelPackages = pkgs.linuxPackages_grsec_desktop_4_5;
```
and run
```
nixos-rebuild switch
```

## Technical details

* System: 16.03.1083.d745044 (Emu)
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: "16.03.1083.d745044"

--- a/nixos/release.nix
+++ b/nixos/release.nix
@@ -213,7 +213,6 @@ in rec {
   #tests.gitlab = callTest tests/gitlab.nix {};
   tests.gnome3 = callTest tests/gnome3.nix {};
   tests.gnome3-gdm = callTest tests/gnome3-gdm.nix {};
-  tests.grsecurity = callTest tests/grsecurity.nix {};
   tests.i3wm = callTest tests/i3wm.nix {};
   tests.installer = callSubTests tests/installer.nix {};
   tests.influxdb = callTest tests/influxdb.nix {};

--- a/pkgs/os-specific/linux/kernel/linux-grsecurity-4.5.nix
+++ b/pkgs/os-specific/linux/kernel/linux-grsecurity-4.5.nix
@@ -1,5 +1,7 @@
 { stdenv, fetchurl, perl, buildLinux, ... } @ args:
 
+throw "grsecurity is unsupported on this release"
+
 import ./generic.nix (args // rec {
   version = "4.5.6";
   extraMeta.branch = "4.5";

